### PR TITLE
Check if repository dir doesn't already exists when cloning

### PR DESCRIPTION
**Pull request by lresende**
When trying to clone a repository, if the repository
directory already exists the operation was causing
JupyterLab to hang. With these changes, an error
message is sent if the directory already exists.

### CODE REVIEW

**Comment by lresende** [comment]
This should address the hanging when the repository is already cloned (or folder already exists) as described in #528

**Comment by telamonian** [comment]
I don't love this approach:

- If the directory already exists *but is empty*, then `git clone` is a valid op
- Why is our gui hanging in this case in the first place? On the command line you do get

    ```
    fatal: destination path 'jupyterlab-git' already exists and is not an empty directory.
    ```

    What's the actual root cause of the problem here?

File: jupyterlab_git/git.py [hunk 1]
```diff
@@ -170,6 +170,13 @@ def clone(self, current_path, repo_url, auth=None):
         :param auth: OPTIONAL dictionary with 'username' and 'password' fields
         :return: response with status code and error message.
         """
+
+        # Validate folder don't already exists
```
Comment (kgryte) [inline jupyterlab_git/git.py]:
```suggestion
        # Validate that the folder doesn't already exist
```

**Comment by telamonian** [comment]
@lresende I fixed up #556 so that it should solve the underlying issue causing the "freeze on clone existing" for everyone. I'd prefer to take the approach in that PR (correctly pass along the error message from git itself) rather than the approach in this PR (add our own pre-emptive error handling). Can you please review #556 and confirm that it solves the same problem you were trying to solve here?

**Comment by lresende** [comment]
This is being taken care by #556

**Pull request closed without merging**